Analytics/Classes/SEGAnalytics.m

```diff
@@ -99,13 +97,24 @@ - (void)dealloc
 NSString *const SEGBuildKeyV1 = @"SEGBuildKey";
 NSString *const SEGBuildKeyV2 = @"SEGBuildKeyV2";
 
-- (void)trackApplicationLifecycleEvents:(BOOL)trackApplicationLifecycleEvents
+- (void)handleAppStateNotification:(NSNotification *)note
 {
-    if (!trackApplicationLifecycleEvents) {
-        return;
+    SEGApplicationLifecyclePayload *payload = [[SEGApplicationLifecyclePayload alloc] init];
+    payload.notificationName = note.name;
+    [self run:SEGEventTypeApplicationLifecycle payload:payload];
+    
+    if ([note.name isEqualToString:UIApplicationDidFinishLaunchingNotification]) {
+        [self _applicationDidFinishLaunchingWithOptions:note.userInfo];
+    } else if ([note.name isEqualToString:UIApplicationWillEnterForegroundNotification]) {
+        [self _applicationWillEnterForeground];
     }
+}
 
-    // Previously SEGBuildKey was stored an integer. This was incorrect because the CFBundleVersion
+- (void)_applicationDidFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    if (!self.configuration.trackApplicationLifecycleEvents) {
+        return;
+    }
+        // Previously SEGBuildKey was stored an integer. This was incorrect because the CFBundleVersion
```

Comment:
@f2prateek @tonyxiao Just pushed up updates to indentation. I used Xcode's default to reindent.  Let me know if it doesn't match up with what you are seeing with spacecommander.